docs/proposals/loadbalancing.md

```diff
@@ -0,0 +1,338 @@
+  * It is possible to run multiple websites on the same loadbalancer ip, embedding everything in the service makes this harder.
+
+2. Admin provisions single/static set of loadbalancers: Cloud loadbalancers have seemingly artificial limitations that prevent just one loadbalancer from being generally useful for all Ingress Paths. Going this way would mean disallowing certain Ingress paths based on the cloud provider.
+
+__Ingress paths__
+
+1. Routes: A pathmap is conceptually inline with how people view the problem, and allows us to perform atomic switch over for blue green deployments. We can still expose a "route like" update interface so users don't have send the entire pathmap to change a single url endpoint.
+
+2. GCEs UrlMap: This resource is way too confusing. It contains a list of host rules each of which can point to a list of path matchers, each of which has a list of path regexes. Users end up bearing the brunt of this complexity.
+
+3. Load balancer resource: This resource would encapsulate all the information in claims and ingress paths. It will get unmanageable in deployments where a single global loadbalancer is claiming all ingress paths. This also doesn't provide a clean abstraction between acquirigin an IP and loadbalancing (the user impedance mismatch problem), while in theory we can have a non-loadbalancer fulfill a claim for an ip, or prepopulate a claim with a known ip just to have services reference it.
+
+__Claims__
+
+1. Have a global loadbalancer claim Ingress paths till it can claim no more: If we decide that users don't get to request new public ips, we could just dissolve the claim into the ingress paths and make the loadbalancer controllers smarter.
+  * Users want to use an ip already in DNS for the service
+  * The controllers need to become smarter of cloudprovider limitations like only using a single cert per loadbalancer ip
+  * Detecting when a single loadbalancer is saturated can be hard
+
+### Issues
+
+What follows is a digest of some of the problems brought up in various github issues. Expect these to make it into the proposal above in due time.
+
+#### How does this work at L4?
+
+People choose L4 for a few different reasons:
+- Cost
+- Latency
+- Throughput of loadbalancer is the bottleneck
+- Cloud provider limitations
+- Want SSL passthrough or SNI switch for multi-certs
+- Need client IP and cannot set loadbalancer proxy as default gateway
+
+Point being, we can't reliably infer all these use case with Service.Type=LoadBalancer or the absence of a UrlMap in the IngressPoint (because they might want L7 for session persistence). Instead, anything serving ingress traffic needs an ingress path (which is why it isn't called urlMap). An L4 ingress path might look like:
+
+```yaml
+apiVersion: v1
+kind: IngressPoint
+metadata:
+  name: l4ingress
+  type: network
+spec:
+  host: www.example.com
```

Comment:
I think what's important at L4 that we don't have today is associating multiple services to a single ip, and different ports. So if we said, the claim is just an ip:port, if you don't specify either you'll get something random, if you specify both it might get rejected, we can have the service reference the claim directly and move `type` into the claim. The service can still exist if the claim is rejected. 

Lets us defer the l4 ingress point till someone actually asks for it.